File: SDKs/amazonaws.com/codecommit/2015-04-13/java/lib/src/main/java/openapisdk/models/shared/Folder.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Folder
 * Returns information about a folder in a repository.
**/
public class Folder {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("absolutePath")
    public String absolutePath;
    public Folder withAbsolutePath(String absolutePath) {
        this.absolutePath = absolutePath;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("relativePath")
    public String relativePath;
    public Folder withRelativePath(String relativePath) {
        this.relativePath = relativePath;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("treeId")
    public String treeId;
    public Folder withTreeId(String treeId) {
        this.treeId = treeId;
        return this;
    }
}

File: SDKs/amazonaws.com/codecommit/2015-04-13/python/src/sdk/models/shared/folder.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class Folder:
    r"""Folder
    Returns information about a folder in a repository.
    """
    
    absolute_path: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('absolutePath') }})
    relative_path: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('relativePath') }})
    tree_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('treeId') }})
    
